packages/next/src/client/dev/dev-build-indicator/internal/dev-build-indicator.ts

```diff
@@ -1,26 +1,36 @@
-/* eslint-disable @typescript-eslint/no-use-before-define */
-import { HMR_ACTIONS_SENT_TO_BROWSER } from '../../server/dev/hot-reloader-types'
-import type { HMR_ACTION_TYPES } from '../../server/dev/hot-reloader-types'
-import { addMessageListener } from '../components/react-dev-overlay/pages/websocket'
-
 type VerticalPosition = 'top' | 'bottom'
-type HorizonalPosition = 'left' | 'right'
 
-export interface ShowHideHandler {
-  show: () => void
-  hide: () => void
-}
+type HorizonalPosition = 'left' | 'right'
 
-export default function initializeBuildWatcher(
-  toggleCallback: (handlers: ShowHideHandler) => void,
-  position = 'bottom-right'
-) {
+const NOOP = () => {}
+
+/**
+ * Shows the build indicator in the browser window.
+ * Used to indicate that Next.js is compiling changes.
+ * This is a singleton function that requires initializeDevBuildIndicator to be called first.
+ */
+export let showDevBuildIndicator = NOOP
```

Comment:
I feel it makes more sense to group those handlers like before, or potentially could have states in the future. And we mutate that instance's property instead of mutating exports. Some bundler could have some protection on the exports, but maybe this is not the most important reason.